src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -291,14 +291,21 @@ class TooltipHTMLContent {
         (el as any).domBelongToZr = true;
         this.el = el;
         const zr = this._zr = api.getZr();
-        const appendToBody = this._appendToBody = opt && opt.appendToBody;
 
-        makeStyleCoord(this._styleCoord, zr, appendToBody, api.getWidth() / 2, api.getHeight() / 2);
-
-        if (appendToBody) {
-            document.body.appendChild(el);
+        let customContainer: HTMLElement | null
+        if (opt && opt.appendToBody) {
+            customContainer = this._customContainer = document.body
+        } else if (opt && opt.getAppendElement) {
```

Comment:
现在container这个变量名被api.getDom()这个入参占用了。

您的意思是把TooltipHTMLContent入参中的api.getDom()移除。然后customContainer改用container这个变量名，并在使用原本container的场合用api.getDom替换是吗？

Reply:
I would suggest providing an option when init:

```
echarts.init(dom, theme, {
    tooltipContainer: '#tooltip-container'
})
```

Reply:
> I would suggest providing an option when init:
> 
> ```
> echarts.init(dom, theme, {
>     tooltipContainer: '#tooltip-container'
> })
> ```

@Ovilia I prefer directly adding it in the `tooltip` option, just like current `appendToBody`. And `appendToBody` can be replaced by `teleport: document.body` after it's supported. Also, it can allow the developers to update the container if they need, which is impossible when specifying in the `init` option. 

As for the naming of tooltip container option, besides `teleport` (figurative, the same as Vue's teleport), `container` / `tooltipContainer` / `appendTo` (more intutive) looks good to me.

Reply:
> besides

Thanks for your reply. I updated my code base on that. 

I am not sure that I got u correctly about the part of `container`. In `TooltipHTMLContent`, the word `container ` was  used to describe the container of the canvas. It might confuse people if we change it to the container fo the tooltip element.

Also, I am poor at naming stuffs. All those names are good by me.

Reply:
@plainheart Thanks for the information about `teleport`, and yes I think it should be better be in the tooltip option than `init`. As about the naming, I thing `appendTo` is a better name and `appendToBody` can be replaced as `appendTo: document.body` or `appendTo: 'body'` (using query selector).

Reply:
> @plainheart Thanks for the information about `teleport`, and yes I think it should be better be in the tooltip option than `init`. As about the naming, I thing `appendTo` is a better name and `appendToBody` can be replaced as `appendTo: document.body` or `appendTo: 'body'` (using query selector).


Thanks.
I updated my pr. Now the config name is `appendTo` and it's value can be a 

- string : a string used by query selector  such as 'body', ‘.scroll-container’ and so on
- function: a function to local a element like `(e) => e.closest('.scroll-container')`, in which the input `e` is the chart container
- domElement: a element for tooltip